persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/SyncDAO.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.persistence.dao
+
+import com.wire.kalium.persistence.SyncQueries
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.CoroutineContext
+
+interface SyncDAO {
+
+    /**
+     * @return [List] of [UserIDEntity] of all other users.
+     * the list does not contain self user ID
+     */
+    suspend fun allOtherUsersId(): List<UserIDEntity>
+}
```

Comment:
This might be a sync-related query. But there's no such thing as a `SyncEntity` or a `Sync` table.

Also, the database itself has no concept of what Sync is. 

It also feels a bit weird to me, that if we ever have another legit use-case where we need all users except the self-user, and it would be in some "sync" umbrella.

It looks like `UserDAO` to me.

I'd also question the need to make a query just for this.

Couldn't we just filter it after the query? We're talking about something that happens every full moon, and would take a millisecond to iterate through 1000 users and eliminate self user from it.

Sure, now that it's done... we can stick with it. But it being outside `UserDAO` is weird to me.